### PR TITLE
Migrate `DecisionState` for multi-tenancy

### LINKED ISSUE #14426: Migrate `DecisionState` for multi-tenancy

### PR DESCRIPTION
**Pull request by korthout**
## Description

Migrates the `DecisionState` for multi-tenancy.

- `DEPRECATED_DMN_DECISIONS` -> `DMN_DECISIONS`
    - Prefix tenant to key
    - Set tenant in value (`PersistedDecision`)
- `DEPRECATED_DMN_DECISION_REQUIREMENTS` -> `DMN_DECISION_REQUIREMENTS`
    - Prefix tenant to key
    - Set tenant in value (`PersistedDecisionRequirements`)
- `DEPRECATED_DMN_LATEST_DECISION_BY_ID` -> `DMN_LATEST_DECISION_BY_ID`
    - Prefix tenant to key
    - Prefix tenant to key in value (Foreign key to the Decision key)
- `DEPRECATED_DMN_LATEST_DECISION_REQUIREMENTS_BY_ID` -> `DMN_LATEST_DECISION_REQUIREMENTS_BY_ID`
    - Prefix tenant to key
    - Prefix tenant to key in value (Foreign key to Decision Requirements key)
- `DEPRECATED_DMN_DECISION_KEY_BY_DECISION_REQUIREMENTS_KEY` -> `DMN_DECISION_KEY_BY_DECISION_REQUIREMENTS_KEY`
    - The key for this CF is a composite key. The tenant must be prefixed to both pats of the composite key.
- `DEPRECATED_DMN_DECISION_KEY_BY_DECISION_ID_AND_VERSION` -> `DMN_DECISION_KEY_BY_DECISION_ID_AND_VERSION`
    - Prefix tenant to key
    - Prefix tenant to key in value (Foreign key to Decision key)
- `DEPRECATED_DMN_DECISION_REQUIREMENTS_KEY_BY_DECISION_REQUIREMENT_ID_AND_VERSION` -> `DMN_DECISION_REQUIREMENTS_KEY_BY_DECISION_REQUIREMENT_ID_AND_VERSION`
    - Prefix tenant to key
    - Prefix tenant to key in value (Foreign key to Decision Requirements key)

## Related issues

closes #14426 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by remcowesterhoud** [comment]
1k lines added 😭 I'm sure it's straightforward though.

**Comment by korthout** [comment]
> 1k lines added 😭 I'm sure it's straightforward though.

Copy paste mostly

**Review by remcowesterhoud — APPROVED** [review]
Great work @korthout. I find these migration surprisingly elegant 🤩 Especially with the comments explaining what they achieve.

bors merge

**Comment by remcowesterhoud** [comment]
Some maven issue, let's retry.

bors retry

File: engine/src/main/java/io/camunda/zeebe/engine/state/migration/to_8_3/legacy/LegacyDecisionState.java
[the remark this answers is not in the document]
Comment (korthout) [reply engine/src/main/java/io/camunda/zeebe/engine/state/migration/to_8_3/legacy/LegacyDecisionState.java:1]: I was even considering refactoring the Process State Migrations :D

**Pull request merged**